### PR TITLE
digidoc: Replace wrap with rpath addition

### PR DESCRIPTION
**Pull request by klemensn**
- libdigidocpp: Replace wrap with rpath addition
- qdigidoc: Explain why LD_LIBRARY_PATH is required

@jagajaga @mmahut @sowelisuwi @flokli

### CODE REVIEW

**Comment by klemensn** [comment]
See https://github.com/open-eid/cmake/pull/35#issuecomment-1148406694

**Comment by flokli** [comment]
I think we should still keep the link to the upstream issue / PR.

Any idea on why you need to patch both the `.so` and the executable?

**Comment by flokli** [comment]
I think we should still keep the link to the upstream issue / PR.

Any idea on why you need to patch both the `.so` and the executable?

**Comment by klemensn** [comment]
> I think we should still keep the link to the upstream issue / PR.

Alright.

> Any idea on why you need to patch both the `.so` and the executable?

Because `qdigidoc4` itself has code to load `opensc-pkcs11.so`, just like `libdigidocpp.so` does.

**Comment by flokli** [comment]
@klemensn can you push the comment additions? Otherwise this LGTM.

**Comment by klemensn** [comment]
> @klemensn can you push the comment additions? Otherwise this LGTM.

I had to correct the approach for qdigidoc4(1), sorry for the premature PR, but it's finished now.
I also rebased after your https://github.com/NixOS/nixpkgs/pull/175991.

While fixing the nixpkgs port I'm also porting the digidoc tools to OpenBSD.
Fixing module loading there I noticed that dlopen(3) behaves differently across systems.

tl;dr: `LD_LIBRARY_PATH` can be avoided for libdigidocpp.so and digidoc-tool(1) on both systems,
but qdigidoc4(1) still requires it on Linux.

Both `qdigidoc4` and `libdigidoc-tool` work as expected with this PR.
